RecoTracker/FinalTrackSelectors/plugins/DuplicateTrackMerger.cc

```diff
@@ -28,6 +32,18 @@
 
 #include "CondFormats/EgammaObjects/interface/GBRForest.h"
 
+// Having this macro reduces the need to pollute the code with
+// #ifdefs. The idea is that the condition is checked only if
+// debugging is enabled. That way the condition expression may use
+// variables that are declared only if EDM_ML_DEBUG is enabled. If it
+// is disabled, rely on the fact that LogTrace should compile to
+// no-op.
+#ifdef EDM_ML_DEBUG
+#define IfLogTrace(cond, cat) if(cond) LogTrace(cat)
+#else
+#define IfLogTrace(cond, cat) LogTrace(cat)
+#endif
```

Comment:
BTW, are there any objections for generalizing this macro (e.g. moving to `FWCore/MessageLogger/interface/MessageLogger.h`)?

This is already a second case where I use it (first one was in [RecoTracker/TkSeedGenerator/plugins/MultiHitGeneratorFromChi2.cc](https://github.com/cms-sw/cmssw/blob/CMSSW_9_0_X/RecoTracker/TkSeedGenerator/plugins/MultiHitGeneratorFromChi2.cc#L46-L56)), and I'm starting to find the pattern useful also elsewhere to get detailed printouts only for specific cases.

@Dr15Jones